File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_CSharp/comum/Rotinas.cs
/*
 * Created by SharpDevelop.
 * User: henrique.souza
 * Date: 10/12/2014
 * Time: 17:42
 * 
 * To change this template use Tools | Options | Coding | Edit Standard Headers.
 */
using System;
using System.IO;
using System.Data;
using System.Data.SQLite;
using System.Drawing;
using System.Drawing.Imaging;
using System.Collections.Generic;
using HFSGuardaDiretorio.objetos;

namespace HFSGuardaDiretorio.comum
{
	/// <summary>
	/// Description of Rotinas.
	/// </summary>
	public static class Rotinas
	{
	    private static SQLiteConnection db;
	
	    public static readonly string HFSGUARDADIR = "HFSGuardaDir";
	
	    public static readonly string BARRA_NORMAL = "\\";
	
	    public static readonly string BARRA_INVERTIDA = "/";
	    
	    public static readonly string FORMATO_DATA = "dd/MM/yyyy";
	    
	    public static readonly string FORMATO_DATAHORA = "dd/MM/yyyy HH:mm:ss";
	    
	    public static readonly string FORMATO_DHARQUIVO = "yyy-MM-dd_HH_mm_ss";
	
	    public static string EXTENSAO_BMP = "bmp";
	    
	    public static string EXTENSAO_ICO = "ico";
	    
	    public static string EXTENSAO_GIF = "gif";
	    
	    public static string EXTENSAO_JPEG = "jpg";
	    
	    public static string EXTENSAO_PNG = "png";
	    
	    public static string EXTENSAO_TIFF = "tif";		
	    
	    public static string FILTRO_TODOS = "Todos os Arquivos (*.*)|*.*";
	    
	    public static string FILTRO_EXPORTAR = "Arquivo TXT (*.txt)|*.txt|Arquivo CSV (*.csv)|*.csv|Arquivo HTML (*.html)|*.html|Arquivo XML (*.xml)|*.xml|Arquivo SQL (*.sql)|*.sql";
	    
	    public static string FILTRO_IMAGEM = "Arquivo BMP (*.bmp)|*.bmp|Arquivo ICO (*.ico)|*.ico|Arquivo GIF (*.gif)|*.gif|Arquivo JPEG (*.png)|*.png|Arquivo PNG (*.jpg)|*.jpg|Arquivo TIFF (*.tif)|*.tif";
	    			
	    public static int LastDelimiter(string subtexto, string texto) {
	        return texto.LastIndexOf(subtexto) + 1;
	    }
	
	    public static int LastDelimiter(char subtexto, string texto) {
	        return texto.LastIndexOf(subtexto) + 1;
	    }
	
	    public static int Pos(string subtexto, string texto) {
	        return texto.IndexOf(subtexto) + 1;
	    }
	
	    public static int Pos(char subtexto, string texto) {
	        return texto.IndexOf(subtexto) + 1;
	    }
	    
	    public static bool StartsStr(string subtexto, string texto) {
	        return texto.StartsWith(subtexto);
	    }
	
	    public static bool ContainsStr(string texto, string subtexto) {
	        return (Pos(subtexto, texto) > 0);
	    }
	    
	    public static bool EndsStr(string subtexto, string texto) {
	        return (LastDelimiter(subtexto, texto) > 0);
	    }
	
	    public static string SubString(string texto, int inicio, int tamanho) {
			if (tamanho > (texto.Length-inicio)){
				tamanho = texto.Length - inicio + 1;
			}
	        return texto.Substring(inicio - 1, tamanho);
	    }
	
	    public static string StringReplaceAll(string texto, string velho, 
	            string novo){
	        return texto.Replace(velho, novo);
	    }
	    
	    public static string QuotedStr(string texto){
	        return "'" + texto + "'";
	    }
	
	    public static string QuotedStr(char texto){
	        return "'" + texto + "'";
	    }
	    
	    public static string ExtractFilePath(string sCaminho) {
	        string retorno;
	
	        FileInfo fileInfo = new FileInfo(sCaminho);
	        retorno = fileInfo.DirectoryName;
	
	        return retorno;
	    }
	
	    public static string ExtractFileName(string sCaminho) {
	        string retorno;
	
	        FileInfo fileInfo = new FileInfo(sCaminho);
	        retorno = fileInfo.Name;
	
	        return retorno;
	    }
	
	    public static bool FileExists(string sNomeArquivo) {
	        FileInfo fileInfo = new FileInfo(sNomeArquivo);
	        return fileInfo.Exists;
	    }
	    
	    public static bool DirectoryExists(string sNomeArquivo) {
	        DirectoryInfo dirInfo = new DirectoryInfo(sNomeArquivo);
	        return dirInfo.Exists;
	    }
	
	    public static void DeleteFile(string sNomeArquivo) {
	        FileInfo fileInfo = new FileInfo(sNomeArquivo);
	        fileInfo.Delete();
	    }
	
	    public static void SaveToFile(byte[] imagem, string sNomeArquivo) {
			FileStream fileStream = null;            
			try {        
				fileStream = new FileStream(sNomeArquivo, 
					FileMode.OpenOrCreate, FileAccess.Write);
				fileStream.Write(imagem, 0, imagem.Length);      
				fileStream.Flush();
			} catch (FileNotFoundException ioEx) {	            
				Console.WriteLine(ioEx.Message);				
			} finally {
				if( fileStream != null ) 
					fileStream.Close();
			}	        
	    }
	
	    public static byte[] LoadFromFile(string sNomeArquivo) {
	    	byte[] imagem = null;
			FileStream fileStream = null;     
			
			try {        
				fileStream = new FileStream(sNomeArquivo, 
					FileMode.Open, FileAccess.Read);

				imagem = new byte[fileStream.Length];
				int numBytesToRead = (int)fileStream.Length;
	            int numBytesRead = 0;
	            while (numBytesToRead > 0)
	            {
	                int n = fileStream.Read(imagem, numBytesRead, numBytesToRead);
	                if (n == 0)
	                    break;	
	                numBytesRead += n;
	                numBytesToRead -= n;
	            }        		
        		
			} catch (FileNotFoundException ioEx) {	            
				Console.WriteLine(ioEx.Message);
			} finally {
				if( fileStream != null ) 
					fileStream.Close();
			}			
	
	        return imagem;
	    }
	
	    public static string formataDate(string formato, DateTime dt) {
	        string ds = "";
	        
	        if (dt != null) {	            
	            ds = dt.ToString(formato);
	        }
	        return ds;
	    }
	
	    public static DateTime StringToDate(string sData) {
	        DateTime data = DateTime.Now;
	        
	        if (sData.Trim().Length > 0) {
	            try {
	                if (sData.Trim().Length == 10)
	                    data = DateTime.ParseExact(sData, FORMATO_DATA,
                                       System.Globalization.CultureInfo.InvariantCulture);
	                else
	                    data = DateTime.ParseExact(sData, FORMATO_DATAHORA,
                                       System.Globalization.CultureInfo.InvariantCulture);
	                    
	            } catch (Exception ex) {
	        		Console.WriteLine("Erro na rotina StringToDate: "
	                        +ex.Message);
	            }
	        }
	        return data;
	    }
	
	    public static string FormatLong(string formato, long valor) {
	        return valor.ToString(formato);
	    }

	    public static string FormatDecimal(decimal valor) {
	        return valor.ToString("#,##0.00");
	    }
	    
	    public static string testaNull(string res) {
	        return (res == null ? "" : res.Trim());
	    }
		
	    public static SQLiteConnection getConexao() {
	        return db;
	    }	
	
	    public static void Conectar(ConexaoParams cp) {
			try	{
				db = new SQLiteConnection("Data Source="+cp.Nome+";Version=3;");
				db.Open();
	
			} catch (Exception ex) {
				Console.WriteLine(ex.Message);
			}
	    }
	
	    public static void Desconectar() {
	        Desconectar(db);
	    }
	
	    public static void Desconectar(SQLiteConnection con) {
	        if (con != null) {
	            con.Close();
	        }
	    }
	
	    public static string trocaSeparador(string texto) {
	        string res = "";
	        if (texto != null && texto.Length > 0) {
	            char[] pedacos = texto.ToCharArray();
	            for (int i = 0; i < pedacos.Length; i++) {
	            	if (pedacos[i] == BARRA_NORMAL[0]) {
	            		pedacos[i] = BARRA_INVERTIDA[0];
	                }
	            }
	            res = new string(pedacos);
	        }
	        return res;
	    }
	    
		public static byte[] imageToByteArray(Image imageIn, ImageFormat formato){
			MemoryStream ms = new MemoryStream();
			imageIn.Save(ms, formato);
			return  ms.ToArray();
		}	

		public static Image byteArrayToImage(byte[] byteArrayIn)
		{
		     MemoryStream ms = new MemoryStream(byteArrayIn);
		     Image returnImage = Image.FromStream(ms);
		     return returnImage;
		}
		
		public static byte[] LerArquivoImagem(string sCaminho)
		{
			byte[] imagebytes = null; 
			FileStream fs = new FileStream(sCaminho, FileMode.Open, FileAccess.Read);
			BinaryReader br = new BinaryReader(fs);
			imagebytes = br.ReadBytes(1000000);
			
			//imagebytes.GetLength(0)
			
			return imagebytes;
		}		

		public static Bitmap byteArrayToBitmap(byte[] byteArrayIn)
		{
		     MemoryStream ms = new MemoryStream(byteArrayIn);
		     Bitmap bmp = new Bitmap(ms);
		     return bmp;
		}

		public static byte[] bitmapToByteArray(Bitmap bmp){
			MemoryStream ms = new MemoryStream();
			bmp.Save(ms, ImageFormat.Bmp);
			return  ms.ToArray();
		}	
		
		
		private static ImageCodecInfo GetEncoder(ImageFormat format)
	    {
	        ImageCodecInfo[] codecs = ImageCodecInfo.GetImageDecoders();
	
	        foreach (ImageCodecInfo codec in codecs)
	        {
	            if (codec.FormatID == format.Guid)
	            {
	                return codec;
	            }
	        }
	
	        return null;
	    }	
			
		public static void SaveJpeg(Image img, string filePath, long quality)
	    {
	        EncoderParameters encoderParameters = new EncoderParameters(1);
	        encoderParameters.Param[0] = new EncoderParameter(Encoder.Quality, quality);
	        img.Save(filePath, GetEncoder(ImageFormat.Jpeg), encoderParameters);
	    }
	
	    public static void SaveJpeg(Image img, Stream stream, long quality)
	    {
	        EncoderParameters encoderParameters = new EncoderParameters(1);
	        encoderParameters.Param[0] = new EncoderParameter(Encoder.Quality, quality);
	        img.Save(stream, GetEncoder(ImageFormat.Jpeg), encoderParameters);
	    }
    
	    public static byte[] BmpParaImagem(byte[] imagemBMP, String formato) {
	        
	        byte[] imagemSaida = null;

	        EncoderParameters encoderParameters = new EncoderParameters(1);
	        encoderParameters.Param[0] = new EncoderParameter(Encoder.Quality, 100L);
	        
	        MemoryStream ms = new MemoryStream();	        
	        Image img = byteArrayToImage(imagemBMP);
	        
	        if (formato.Equals(EXTENSAO_BMP))
				img.Save(ms, GetEncoder(ImageFormat.Bmp), encoderParameters);
	        else if (formato.Equals(EXTENSAO_JPEG))
				img.Save(ms, GetEncoder(ImageFormat.Jpeg), encoderParameters);
			else if (formato.Equals(EXTENSAO_ICO))
				img.Save(ms, GetEncoder(ImageFormat.Icon), encoderParameters);
			else if (formato.Equals(EXTENSAO_GIF))
				img.Save(ms, GetEncoder(ImageFormat.Gif), encoderParameters);				
			else if (formato.Equals(EXTENSAO_PNG))
				img.Save(ms, GetEncoder(ImageFormat.Png), encoderParameters);
			else if (formato.Equals(EXTENSAO_TIFF))
				img.Save(ms, GetEncoder(ImageFormat.Tiff), encoderParameters);
				
	        imagemSaida = ms.ToArray();
						
	        return imagemSaida;
	    }

		public static List<FileInfo> listFiles(string sCaminho) 
		{
			List<FileInfo> lista = new List<FileInfo>();
			foreach (string dir in Directory.GetDirectories(sCaminho)) {
				lista.Add(new FileInfo(dir));
			}
			foreach (string arq in Directory.GetFiles(sCaminho)) {
				lista.Add(new FileInfo(arq));
			}
			return lista;
		}
		
	}
}

File: HFSGuardaDiretorioVS2022/HFSGuardaDiretorio_VBNet/comum/Rotinas.vb
'
' * Created by SharpDevelop.
' * User: henrique.souza
' * Date: 10/12/2014
' * Time: 17:42
' * 
' * To change this template use Tools | Options | Coding | Edit Standard Headers.
' 

Imports System.IO
Imports System.Data
Imports System.Data.SQLite
Imports System.Drawing
Imports System.Drawing.Imaging
Imports System.Collections.Generic
Imports HFSGuardaDiretorio.objetos

Namespace comum
	''' <summary>
	''' Description of Rotinas.
	''' </summary>
	Public NotInheritable Class Rotinas
		Private Sub New()
		End Sub
		Private Shared db As SQLiteConnection

		Public Shared ReadOnly HFSGUARDADIR As String = "HFSGuardaDir"

		Public Shared ReadOnly BARRA_NORMAL As String = "\"

		Public Shared ReadOnly BARRA_INVERTIDA As String = "/"

		Public Shared ReadOnly FORMATO_DATA As String = "dd/MM/yyyy"

		Public Shared ReadOnly FORMATO_DATAHORA As String = "dd/MM/yyyy HH:mm:ss"

		Public Shared ReadOnly FORMATO_DHARQUIVO As String = "yyy-MM-dd_HH_mm_ss"

		Public Shared EXTENSAO_BMP As String = "bmp"

		Public Shared EXTENSAO_ICO As String = "ico"

		Public Shared EXTENSAO_GIF As String = "gif"

		Public Shared EXTENSAO_JPEG As String = "jpg"

		Public Shared EXTENSAO_PNG As String = "png"

		Public Shared EXTENSAO_TIFF As String = "tif"

		Public Shared FILTRO_TODOS As String = "Todos os Arquivos (*.*)|*.*"

		Public Shared FILTRO_EXPORTAR As String = "Arquivo TXT (*.txt)|*.txt|Arquivo CSV (*.csv)|*.csv|Arquivo HTML (*.html)|*.html|Arquivo XML (*.xml)|*.xml|Arquivo SQL (*.sql)|*.sql"

		Public Shared FILTRO_IMAGEM As String = "Arquivo BMP (*.bmp)|*.bmp|Arquivo ICO (*.ico)|*.ico|Arquivo GIF (*.gif)|*.gif|Arquivo JPEG (*.png)|*.png|Arquivo PNG (*.jpg)|*.jpg|Arquivo TIFF (*.tif)|*.tif"

		Public Shared Function LastDelimiter(subtexto As String, texto As String) As Integer
			Return texto.LastIndexOf(subtexto) + 1
		End Function

		Public Shared Function LastDelimiter(subtexto As Char, texto As String) As Integer
			Return texto.LastIndexOf(subtexto) + 1
		End Function

		Public Shared Function Pos(subtexto As String, texto As String) As Integer
			Return texto.IndexOf(subtexto) + 1
		End Function

		Public Shared Function Pos(subtexto As Char, texto As String) As Integer
			Return texto.IndexOf(subtexto) + 1
		End Function

		Public Shared Function StartsStr(subtexto As String, texto As String) As Boolean
			Return texto.StartsWith(subtexto)
		End Function

		Public Shared Function ContainsStr(texto As String, subtexto As String) As Boolean
			Return (Pos(subtexto, texto) > 0)
		End Function

		Public Shared Function EndsStr(subtexto As String, texto As String) As Boolean
			Return (LastDelimiter(subtexto, texto) > 0)
		End Function

		Public Shared Function SubString(texto As String, inicio As Integer, tamanho As Integer) As String
			If tamanho > (texto.Length - inicio) Then
				tamanho = texto.Length - inicio + 1
			End If
			Return texto.Substring(inicio - 1, tamanho)
		End Function

		Public Shared Function StringReplaceAll(texto As String, velho As String, novo As String) As String
			Return texto.Replace(velho, novo)
		End Function

		Public Shared Function QuotedStr(texto As String) As String
			Return "'" & texto & "'"
		End Function

		Public Shared Function QuotedStr(texto As Char) As String
			Return "'" & texto & "'"
		End Function

		Public Shared Function ExtractFilePath(sCaminho As String) As String
			Dim retorno As String

			Dim fileInfo As New FileInfo(sCaminho)
			retorno = fileInfo.DirectoryName

			Return retorno
		End Function

		Public Shared Function ExtractFileName(sCaminho As String) As String
			Dim retorno As String

			Dim fileInfo As New FileInfo(sCaminho)
			retorno = fileInfo.Name

			Return retorno
		End Function

		Public Shared Function FileExists(sNomeArquivo As String) As Boolean
			Dim fileInfo As New FileInfo(sNomeArquivo)
			Return fileInfo.Exists
		End Function

		Public Shared Function DirectoryExists(sNomeArquivo As String) As Boolean
			Dim dirInfo As New DirectoryInfo(sNomeArquivo)
			Return dirInfo.Exists
		End Function

		Public Shared Sub DeleteFile(sNomeArquivo As String)
			Dim fileInfo As New FileInfo(sNomeArquivo)
			fileInfo.Delete()
		End Sub

		Public Shared Sub SaveToFile(imagem As Byte(), sNomeArquivo As String)
			Dim fileStream As FileStream = Nothing
			Try
				fileStream = New FileStream(sNomeArquivo, FileMode.OpenOrCreate, FileAccess.Write)
				fileStream.Write(imagem, 0, imagem.Length)
				fileStream.Flush()
			Catch ioEx As FileNotFoundException
				Console.WriteLine(ioEx.Message)
			Finally
				If fileStream IsNot Nothing Then
					fileStream.Close()
				End If
			End Try
		End Sub

		Public Shared Function LoadFromFile(sNomeArquivo As String) As Byte()
			Dim imagem As Byte() = Nothing
			Dim fileStream As FileStream = Nothing

			Try
				fileStream = New FileStream(sNomeArquivo, FileMode.Open, FileAccess.Read)
				
				imagem = New Byte(fileStream.Length - 1) {}
				Dim numBytesToRead As Integer = CInt(fileStream.Length)
				Dim numBytesRead As Integer = 0
				While numBytesToRead > 0
					Dim n As Integer = fileStream.Read(imagem, numBytesRead, numBytesToRead)
					If n = 0 Then
						Exit While
					End If
					numBytesRead += n
					numBytesToRead -= n

				End While
			Catch ioEx As FileNotFoundException
				Console.WriteLine(ioEx.Message)
			Finally
				If fileStream IsNot Nothing Then
					fileStream.Close()
				End If
			End Try

			Return imagem
		End Function

		Public Shared Function formataDate(formato As String, dt As DateTime) As String
			Dim ds As String = ""

			If dt <> Nothing Then
				ds = dt.ToString(formato)
			End If
			Return ds
		End Function

		Public Shared Function StringToDate(sData As String) As DateTime
			Dim data As DateTime = DateTime.Now

			If sData.Trim().Length > 0 Then
				Try
					If sData.Trim().Length = 10 Then
						data = DateTime.ParseExact(sData, FORMATO_DATA, System.Globalization.CultureInfo.InvariantCulture)
					Else
						data = DateTime.ParseExact(sData, FORMATO_DATAHORA, System.Globalization.CultureInfo.InvariantCulture)

					End If
				Catch ex As Exception
					Console.WriteLine("Erro na rotina StringToDate: " & ex.Message)
				End Try
			End If
			Return data
		End Function

		Public Shared Function FormatLong(formato As String, valor As Long) As String
			Return valor.ToString(formato)
		End Function

		Public Shared Function FormatDecimal(valor As Decimal) As String
			Return valor.ToString("#,##0.00")
		End Function

		Public Shared Function testaNull(res As String) As String
			Return (If(res Is Nothing, "", res.Trim()))
		End Function

		Public Shared Function getConexao() As SQLiteConnection
			Return db
		End Function

		Public Shared Sub Conectar(cp As ConexaoParams)
			Try
				db = New SQLiteConnection("Data Source=" & cp.Nome & ";Version=3;")

				db.Open()
			Catch ex As Exception
				Console.WriteLine(ex.Message)
			End Try
		End Sub

		Public Shared Sub Desconectar()
			Desconectar(db)
		End Sub

		Public Shared Sub Desconectar(con As SQLiteConnection)
			If con IsNot Nothing Then
				con.Close()
			End If
		End Sub

		Public Shared Function trocaSeparador(texto As String) As String
			Dim res As String = ""
			If texto IsNot Nothing AndAlso texto.Length > 0 Then
				Dim pedacos As Char() = texto.ToCharArray()
				For i As Integer = 0 To pedacos.Length - 1
					If pedacos(i) = BARRA_NORMAL(0) Then
						pedacos(i) = BARRA_INVERTIDA(0)
					End If
				Next
				res = New String(pedacos)
			End If
			Return res
		End Function

		Public Shared Function imageToByteArray(imageIn As Image, formato As ImageFormat) As Byte()
			Dim ms As New MemoryStream()
			imageIn.Save(ms, formato)
			Return ms.ToArray()
		End Function

		Public Shared Function byteArrayToImage(byteArrayIn As Byte()) As Image
			Dim ms As New MemoryStream(byteArrayIn)
			Dim returnImage As Image = Image.FromStream(ms)
			Return returnImage
		End Function

		Public Shared Function LerArquivoImagem(sCaminho As String) As Byte()
			Dim imagebytes As Byte() = Nothing
			Dim fs As New FileStream(sCaminho, FileMode.Open, FileAccess.Read)
			Dim br As New BinaryReader(fs)
			imagebytes = br.ReadBytes(1000000)

			'imagebytes.GetLength(0)

			Return imagebytes
		End Function

		Public Shared Function byteArrayToBitmap(byteArrayIn As Byte()) As Bitmap
			Dim ms As New MemoryStream(byteArrayIn)
			Dim bmp As New Bitmap(ms)
			Return bmp
		End Function

		Public Shared Function bitmapToByteArray(bmp As Bitmap) As Byte()
			Dim ms As New MemoryStream()
			bmp.Save(ms, ImageFormat.Bmp)
			Return ms.ToArray()
		End Function


		Private Shared Function GetEncoder(format As ImageFormat) As ImageCodecInfo
			Dim codecs As ImageCodecInfo() = ImageCodecInfo.GetImageDecoders()

			For Each codec As ImageCodecInfo In codecs
				If codec.FormatID = format.Guid Then
					Return codec
				End If
			Next

			Return Nothing
		End Function

		Public Shared Sub SaveJpeg(img As Image, filePath As String, quality As Long)
			Dim encoderParameters As EncoderParameters = New EncoderParameters(1)
			encoderParameters.Param(0) = New EncoderParameter(Encoder.Quality, quality)
			img.Save(filePath, GetEncoder(ImageFormat.Jpeg), encoderParameters)
		End Sub

		Public Shared Sub SaveJpeg(img As Image, stream As Stream, quality As Long)
			Dim encoderParameters As EncoderParameters = New EncoderParameters(1)
			encoderParameters.Param(0) = New EncoderParameter(Encoder.Quality, quality)
			img.Save(stream, GetEncoder(ImageFormat.Jpeg), encoderParameters)
		End Sub

		Public Shared Function BmpParaImagem(imagemBMP As Byte(), formato As [String]) As Byte()

			Dim imagemSaida As Byte() = Nothing

			Dim encoderParameters As EncoderParameters = New EncoderParameters(1)
			encoderParameters.Param(0) = New EncoderParameter(Encoder.Quality, 100L)

			Dim ms As New MemoryStream()
			Dim img As Image = byteArrayToImage(imagemBMP)

			If formato.Equals(EXTENSAO_BMP) Then
				img.Save(ms, GetEncoder(ImageFormat.Bmp), encoderParameters)
			ElseIf formato.Equals(EXTENSAO_JPEG) Then
				img.Save(ms, GetEncoder(ImageFormat.Jpeg), encoderParameters)
			ElseIf formato.Equals(EXTENSAO_ICO) Then
				img.Save(ms, GetEncoder(ImageFormat.Icon), encoderParameters)
			ElseIf formato.Equals(EXTENSAO_GIF) Then
				img.Save(ms, GetEncoder(ImageFormat.Gif), encoderParameters)
			ElseIf formato.Equals(EXTENSAO_PNG) Then
				img.Save(ms, GetEncoder(ImageFormat.Png), encoderParameters)
			ElseIf formato.Equals(EXTENSAO_TIFF) Then
				img.Save(ms, GetEncoder(ImageFormat.Tiff), encoderParameters)
			End If

			imagemSaida = ms.ToArray()

			Return imagemSaida
		End Function

		Public Shared Function listFiles(sCaminho As String) As List(Of FileInfo)
			Dim lista As New List(Of FileInfo)()
			For Each dir As String In Directory.GetDirectories(sCaminho)
				lista.Add(New FileInfo(dir))
			Next
			For Each arq As String In Directory.GetFiles(sCaminho)
				lista.Add(New FileInfo(arq))
			Next
			Return lista
		End Function

	End Class
End Namespace
